extension/src/vscode/quickPick.ts

```diff
@@ -219,6 +220,55 @@ export const quickPickLimitedValues = <T>(
     quickPick.show()
   })
 
+const getUserOrderedSelection = <T>(
+  orderedSelection: T[],
+  newSelection: readonly QuickPickItemWithValue<T>[]
+) => {
+  const itemValues: T[] = []
+  for (const item of newSelection) {
```

Comment:
[C] This function is really dense. Can we break into `addItems`/`removeItems`?

I.e something like
```
itemValues = addNewItems(...)
return removeMissingItems(...)
```

Reply:
(definitely try to improve on the code provided at speed as a comment in a pull request)